server/routes/bookings/cancelVisitController.ts

```diff
@@ -0,0 +1,98 @@
+import type { RequestHandler } from 'express'
+import { Meta, ValidationChain, matchedData, param, body } from 'express-validator'
+import { BookerService, VisitService } from '../../services'
+import paths from '../../constants/paths'
+
+export default class CancelVisitController {
+  public constructor(
+    private readonly bookerService: BookerService,
+    private readonly visitService: VisitService,
+  ) {}
+
+  public confirmCancelView(): RequestHandler {
+    return async (req, res) => {
+      const { booker, bookings } = req.session
+      const { visits } = bookings
+
+      const prisoner = booker.prisoners?.[0]
+        ? booker.prisoners?.[0]
+        : (await this.bookerService.getPrisoners(booker.reference))?.[0]
```

Comment:
Also could this come back with no prisoners? So trying to access `[0]` will cause a fail?

Reply:
Yes, potentially. Have spoken to @hutcheonb-moj about this. We've got the same block of code in a few places now. It's all to do with needing a prisoner's name to display. I've arranged to get a change to the API done (which is in progress) so we get prisoner name returned with the visit data itself so no longer need to fish it out from the booker record in the session.

So, OK for now I think as better, less brittle solution coming immanently!